src/BehaviorRelay.ts

```diff
@@ -11,28 +11,38 @@ import { Subscriber, Subscription } from 'rxjs';
  * @class BehaviorRelay<T>
  */
 export class BehaviorRelay<T> extends Relay<T> {
+  private _hasValue = false;
```

Comment:
I think making `hasValue` as function is better than using it as variable.

Reply:
Sorry for my bad english 😢My previous comment wasn't meaning this.
I think `_hasValue` variable is useless, because passing **non-null value** is forced to `next`.
So, I think we can replace `_hasValue` with 
```ts
hasValue(): boolean {
  return (typeof _value !== 'undefined');
}
```
:D

Reply:
Ah I think I know what you mean! 😃 
However, if I understand you correctly this will not work with the following case if we desire to emit undefined values:
```js
const relay = new BehaviorRelay();
relay.subscribe((val) => console.log(`A: ${val}`));
relay.next(undefined);
relay.subscribe((val) => console.log(`B: ${val}`));
relay.next(123);
```
console output:
```
A: undefined
B: undefined
A: 123
B: 123
```
or what do you think? 😃